Adds the ability to create external plugins

nul800sebastiaan opened the pull request:
See issue https://github.com/Lone-Coder/letsencrypt-win-simple/issues/732

WouterTinus commented on letsencrypt-win-simple/Services/PluginService.cs:
```diff
@@ -102,13 +103,26 @@ public PluginService()
-            if (!allowNull)
+            var baseDir = AppDomain.CurrentDomain.BaseDirectory;
+            string[] extensions = { ".dll" };
+            var allFiles = Directory.EnumerateFileSystemEntries(baseDir, "*.*")
```
Why not filter by *.dll instead of this construction?

nul800sebastiaan replied:
Good point, I'll update!

WouterTinus commented on letsencrypt-win-simple/Services/PluginService.cs:
```diff
@@ -102,13 +103,26 @@ public PluginService()
+            var allFiles = Directory.EnumerateFileSystemEntries(baseDir, "*.*")
+                .Where(x => extensions.Any(ext => ext == Path.GetExtension(x)));
+            var assemblies = new List<Assembly> { Assembly.GetExecutingAssembly() };
+            assemblies.AddRange(allFiles.Select(AssemblyName.GetAssemblyName).Select(Assembly.Load));
```
Maybe add some try-catch with a Log.Warn if it fails to load? (Not sure if the logger is available at this point)

nul800sebastiaan replied:
Yep, agreed, couldn't figure out how to get the logger, but I'll look into it again.

nul800sebastiaan commented:
@WouterTinus Thanks for the review, this is now updated:

- Only looking for dll files without the verbose filterinig
- Added logger (not sure if this is the best way but it seems to make sense
- Loads the internal plugins regardless of anything
- Tries to load plugin types in dlls, with a try-catch to handle problems

WouterTinus commented:
Thanks for the update!
